src/io/parquet/write/mod.rs

```diff
@@ -163,8 +168,29 @@ pub fn array_to_pages(
                     )
                 })
             }
-            _ => array_to_page(array, type_, nested, options, encoding)
-                .map(|page| DynIter::new(std::iter::once(Ok(page)))),
+            _ => {
+                const DEFAULT_PAGE_SIZE: usize = 1024 * 1024;
+                let page_size = options.data_pagesize_limit.unwrap_or(DEFAULT_PAGE_SIZE);
+                let bytes_per_row =
+                    (((array_byte_size as f64) / (array.len() as f64)) as usize).max(1);
+                let rows_per_page = (page_size / bytes_per_row).max(1);
```

Comment:
Maybe we should divide by `bytes_per_row + 1`? Are we certain that we never hit this branch with an empty array?

Reply:
`bytes_per_row + 1` may make `rows_per_page` less than the expect value if bytes_per_row is small.

As here `bytes_per_row` is a float value GTE than 1 and `rows_per_page` is GTE than 1, so if the array is empty, the iterator will be empty.